### PR TITLE
[AIRFLOW-5664] Store timestamps with microseconds precision in GCSToPSQL

### PR DESCRIPTION
**Pull request by osule**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/5664) issues and references them in the PR title. ~For example, "\[AIRFLOW-XXX\] My Airflow PR"~
  - https://issues.apache.org/jira/browse/AIRFLOW-5664
  - ~In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.~
  - ~In case you are proposing a fundamental code change, you need to create an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)).~
  - ~In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).~

### Description

- [x] Here are some details about my PR, ~including screenshots of any UI changes~:

>Microseconds value is lost in the conversion to timestamp using time.mktime.
>Timestamp is now computed to be precise up to microseconds.

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason: Synonymous with testing pendulum fluent helpers. 

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### ~Documentation~

- [x] ~In case of new functionality, my PR adds documentation that describes how to use it.~
  - ~All the public functions and the classes in the PR contain docstrings that explain what it does~
  - ~If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release~

### CODE REVIEW

File: airflow/operators/postgres_to_gcs.py [hunk 1]
```diff
@@ -88,7 +90,8 @@ def convert_type(self, value, schema_type):
         Decimals are converted to floats. Times are converted to seconds.
         """
         if isinstance(value, (datetime.datetime, datetime.date)):
-            return time.mktime(value.timetuple())
+            return pendulum.parse(
+                value.isoformat()).in_tz('UTC').float_timestamp
```
Comment (RosterIn) [inline airflow/operators/postgres_to_gcs.py]: Why UTC?

**Review by RosterIn — COMMENTED** [review]
The title of PR and Jira says `GCSToPSQL` but you suggest edit to `postgres_to_gcs` ?

File: airflow/operators/postgres_to_gcs.py (hunk 1 above)
Comment (osule) [reply airflow/operators/postgres_to_gcs.py]: @RosterIn  At the time I wrote this, I understood that the ISO-8601 string representation for `value` could have a timezone offset and thought extra handling necessary to convert the parsed datetime in the local time.

For a fact, this extra conversion isn't necessary to perform. I'll remove it.

**Comment by osule** [comment]
> The title of PR and Jira says `GCSToPSQL` but you suggest edit to `postgres_to_gcs` ?

@Rosterln I could be wrong. I think it's mistitled.  The description for the Jira ticket refers to the `postgres_to_gcs` operator.

**Comment by RosterIn** [comment]
Looks good

**Review by RosterIn — APPROVED** [review]

**Review by turbaszek — APPROVED** [review]

**Comment by turbaszek** [comment]
@osule can you please, rebase onto new master?

**Comment by osule** [comment]
> @osule can you please, rebase onto new master?

Sure, I'll resolve the conflicts and push a changed commit.

**Comment by RosterIn** [comment]
@nuclearpinguin can this PR get merged?

**Comment by ashb** [comment]
@nuclearpinguin Commit messages are what we use to build our changelog; can you please make sure that they are descriptive for end users to usefully identify the change. For example in this case this commit subject does not identify _where_ this change is -- it should have included "PostgresToGCSOperator" or similar in the subject.

File: airflow/providers/google/cloud/operators/postgres_to_gcs.py [hunk 2]
```diff
@@ -88,7 +90,7 @@ def convert_type(self, value, schema_type):
         Decimals are converted to floats. Times are converted to seconds.
         """
         if isinstance(value, (datetime.datetime, datetime.date)):
-            return time.mktime(value.timetuple())
+            return pendulum.parse(value.isoformat()).float_timestamp
```
Comment (ashb) [inline airflow/providers/google/cloud/operators/postgres_to_gcs.py:93]: Holy in-efficiency batman!

We're doing from a datetime, to a string (a bit slow) parsing it _back_ to a datetime-like object (_VERY_ slow) to then call an attribute on it.

This should be

```
             return value.timestamp()
```

Also the comment should be clarified. "Times are converted to fractional-seconds"

**Comment by nuclearpinguin** [comment]
> @nuclearpinguin Commit messages are what we use to build our changelog; can you please make sure that they are descriptive for end users to usefully identify the change. For example in this case this commit subject does not identify _where_ this change is -- it should have included "PostgresToGCSOperator" or similar in the subject.

Sorry, next time I will check if the commit message is the same as the PR title.

File: airflow/providers/google/cloud/operators/postgres_to_gcs.py (hunk 2 above)
Comment (osule) [reply airflow/providers/google/cloud/operators/postgres_to_gcs.py:93]: You're correct that this is inefficient. 
However, `value.timestamp` doesn't take timezone information into account.

Now that I think of it, it should have been written as  or some other way

```
return pendulum.timezone('UTC').convert(value).timestamp()
```

**Comment by osule** [comment]
@nuclearpinguin Should I open a new PR that addresses concern from the discussion?

Or will you handle it from here?

File: airflow/providers/google/cloud/operators/postgres_to_gcs.py (hunk 2 above)
Comment (turbaszek) [reply airflow/providers/google/cloud/operators/postgres_to_gcs.py:93]: @ashb WDYT? Should the commit be reverted and new one added?

**Pull request merged**